packages/core/src/connectors/walletConnect.ts

```diff
@@ -9,36 +11,49 @@ import type { Chain } from '../types'
 import { normalizeChainId } from '../utils'
 import { Connector } from './base'
 
-/**
- * Wallets that support chain switching through WalletConnect
- * - imToken (token.im)
- * - MetaMask (metamask.io)
- * - Rainbow (rainbow.me)
- * - Trust Wallet (trustwallet.com)
- */
-const switchChainAllowedRegex = /(imtoken|metamask|rainbow|trust wallet)/i
+// Shared config for WalletConnect v2
+const sharedConfig = {
+  namespace: 'eip155',
+}
 
-type WalletConnectOptions = ConstructorParameters<
-  typeof WalletConnectProvider
->[0]
+type WalletConnectOptions = {
+  /** When `true`, uses default WalletConnect QR Code modal */
+  qrcode?: boolean
+} & (
+  | ({
+      version?: '1'
+    } & ConstructorParameters<typeof WalletConnectProvider>[0])
+  | ({
+      /**
+       * Project ID associated with [WalletConnect account](https://cloud.walletconnect.com)
+       */
+      projectId: NonNullable<UniversalProviderOpts['projectId']>
```

Comment:
Isn't it possible to have no projectId if using your own relayer for walletconnect v2?